consensus/src/msg_handler.rs

```diff
@@ -67,70 +80,83 @@ pub trait MsgHandler {
-                                    round_committees,
-                                )?;
-                            }
-                            node_data::message::Payload::Candidate(c) => {
-                                proposal::handler::verify_stateless(
-                                    c,
-                                    round_committees,
-                                )?;
-                            }
-                            node_data::message::Payload::Quorum(_) => {}
-                            node_data::message::Payload::Block(_) => {}
-                            _ => {
-                                warn!(
-                                    "future message not repropagated {:?}",
-                                    msg.topic()
-                                );
-                                Err(ConsensusError::InvalidMsgType)?;
-                            }
-                        }
-                    } else {
-                        warn!("Future committee for iteration {iteration} not generated; skipping pre-verification for {:?} message", msg.topic());
-                    }
+            let step = msg.get_step();
+            if let Some(committee) = round_committees.get_committee(step) {
+                // Ensure the message originates from a committee
+                // member.
+                if !committee.is_member(signer) {
+                    return Err(ConsensusError::NotCommitteeMember);
                 }
 
-                Err(ConsensusError::FutureEvent)
+                match &msg.payload {
+                    node_data::message::Payload::Ratification(_) => {
+                        RatificationHandler::verify_stateless(
+                            msg,
+                            round_committees,
+                        )?;
+                    }
+                    node_data::message::Payload::Validation(_) => {
+                        ValidationHandler::verify_stateless(
+                            msg,
+                            round_committees,
+                        )?;
+                    }
+                    node_data::message::Payload::Candidate(c) => {
+                        proposal::handler::verify_stateless(
+                            c,
+                            round_committees,
+                        )?;
+                    }
+                    node_data::message::Payload::Quorum(_) => {}
+                    node_data::message::Payload::Block(_) => {}
+                    _ => {
+                        warn!(
+                            "{status:?} message not repropagated {:?}",
+                            msg.topic()
+                        );
+                        Err(ConsensusError::InvalidMsgType)?;
+                    }
+                }
+            } else {
+                warn!("{status:?} committee for step {step} not generated; skipping pre-verification for {:?} message", msg.topic());
```

Comment:
Skipping is fine for future messages, as they will be verified later.
But I'm not sure we should just print a warning when verifying emergency-iteration messages

Reply:
Current emergency iteration messages are note verified by this

They are supposed to be verified by the living step

Reply:
I mean past-iteration messages

Reply:
I expect the previous committee to be generated (otherwise it should be a bug)

That's why I added `{status:?}` in the warn message, to properly detect if this happens